domain/application/state/types.go

```diff
@@ -20,6 +25,37 @@ type applicationName struct {
 	Name string `db:"name"`
 }
 
+type applicationDetails struct {
+	ApplicationID string    `db:"uuid"`
+	Name          string    `db:"name"`
+	CharmID       string    `db:"charm_uuid"`
+	LifeID        life.Life `db:"life_id"`
+}
+
+type applicationPlatform struct {
+	ApplicationID  string                   `db:"application_uuid"`
+	OsID           application.OSType       `db:"os_id"`
```

Comment:
As much as it's ugly `OSID` is probably the correct way, but I think it's fine to leave it as is.

Reply:
renamed to OSTypeID